docs/source/developers/crossbow.rst

```diff
@@ -0,0 +1,284 @@
+.. raw:: html
+
+   <!--
+   Licensed to the Apache Software Foundation (ASF) under one
+   or more contributor license agreements.  See the NOTICE file
+   distributed with this work for additional information
+   regarding copyright ownership.  The ASF licenses this file
+   to you under the Apache License, Version 2.0 (the
+   "License"); you may not use this file except in compliance
+   with the License.  You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+   -->
+
+Packaging and Testing with Crossbow
+===================================
+
+The content of this directory aims for automating the historically error
+prone process of Arrow packaging and integration testing.
+
+Packages:
+  - C++ and Python `conda-forge packages`_ for Linux, Mac and Windows
+  - Python `Wheels`_ for Linux, Mac and Windows
+  - C++ and GLib `Linux packages`_ for multiple distributions
+  - Java for Gandiva
+
+Integration tests:
+  - Various docker tests
+  - Pandas
+  - Dask
+  - Turbodbc
+  - HDFS
+  - Spark
+
+Architecture
+------------
+
+Executors
+~~~~~~~~~
+
+Individual jobs are executed on public CI services, currently: - Linux:
+TravisCI - Mac: TravisCI - Windows: AppVeyor
+
+Queue
+~~~~~
+
+Because of the nature of how the CI services work, the scheduling of
+jobs happens through an additional git repository, which acts like a job
+queue for the tasks. A job is a git commit on a particular git branch,
+containing only the required configuration file to run the requested
+build (like ``.travis.yml``, ``appveyor.yml`` or ``azure-pipelines.yml``).
+
+Scheduler
+~~~~~~~~~
+
+`Crossbow.py`_ handles version generation, task rendering and
+submission. The packaging tasks are defined in ``tasks.yml`` and the
+integration tests are configured in ``tests.yml``.
+
+Install
+-------
+
+   The following guide depends on GitHub, but theoretically any git
+   server can be used.
+
+1. `Create the queue repository`_
+
+2. Enable `TravisCI`_, `Appveyor`_, `Azure Pipelines_` and `CircleCI`_
+   integrations on for the newly created queue repository.
+
+   -  turn off Travis’ `auto cancellation`_ feature on branches
+
+3. Clone the newly created, by default the scripts looks for
+   ``crossbow`` next to arrow repository.
+
+   .. code:: bash
+
+      git clone https://github.com/<user>/crossbow crossbow
+
+   **Important note:** Crossbow only supports GitHub token based
```

Comment:
I worked out how to use GitHub deploy keys for arrow-site (though INFRA is blocking their use) so we could add that here, if there's anyone who's uncomfortable using a PAT with broad scope.